llvm/test/Transforms/LoopVectorize/AArch64/force-target-instruction-cost.ll

```diff
@@ -423,12 +413,16 @@ else.2:
 loop.latch:
   %iv.next = add i64 %iv, 1
   %ec = icmp eq i64 %iv, 64
-  br i1 %ec, label %exit, label %loop.header
+  br i1 %ec, label %exit, label %loop.header, !llvm.loop !0
 
 exit:
   ret void
 }
 
+!0 = distinct !{!0, !1}
```

Comment:
How much work would it be to make the relevant instructions (the ones computing the block masks?) not loop-invariant?


Something like 
```
  %gep.1 = getelementptr inbounds i64, ptr %src.1, i64 %iv
  %x = load i64, %gep.1
  %c1 = icmp eq i64 %x, 0
```

Reply:
That seemed to work more or less, I've removed the force vectorization hint too